sql/src/test/java/io/crate/integrationtests/MetricsITest.java

```diff
@@ -72,13 +79,31 @@ public void testSimpleQueryOnMetrics() {
     }
 
     @Test
-    public void testTotalCountOnMetrics() {
+    public void testTotalCountOnMetrics() throws Exception {
+        String stmt = "SELECT 1";
         int numQueries = 100;
         for (int i = 0; i < numQueries; i++) {
-            execute("SELECT 1");
+            execute(stmt);
         }
 
         execute("SELECT sum(total_count) FROM sys.jobs_metrics WHERE classification['type'] = 'SELECT'");
+
+        // We record the statements in the jobsLogs and the histograms **after** we notify the result receivers
+        // (see {@link JobsLogsUpdateListener usage).
+        // So it might happen that the "set global ..." statement execution is returned to this test but the recording
+        // in the log is done AFTER the execution of the statement (because async programming is evil like that).
+        assertBusy(() -> {
```

Comment:
Couldn't we just put the 

```
execute("SELECT sum(total_count) FROM sys.jobs_metrics WHERE classification['type'] = 'SELECT'");
assertThat(response.rows()[0][0], Matchers.is((long) numQueries));
```

in a `assertBusy` block? I don't think we need to encode a "jobsLog" and "jobs_metric" relationship into the test.

Reply:
Hmm. thought of it but then the `SELECT sum(total_count) FROM sys.jobs_metrics...` Statement could potentially be executed multiple times...

Reply:
I agree that putting an relation to jobsLog isn't a nice solution, also I see the problems @matriv describes.
what about checking the histogram via the `JobsLogs.metrics()` on the instances instead?